x/gamm/keeper/pool_service.go

```diff
@@ -57,138 +54,6 @@ func (k Keeper) CalculateSpotPrice(
 	return spotPrice, err
 }
 
-func validateCreatePoolMsg(ctx sdk.Context, msg swaproutertypes.CreatePoolMsg) error {
-	err := msg.Validate(ctx)
-	if err != nil {
-		return err
-	}
-
-	initialPoolLiquidity := msg.InitialLiquidity()
-	numAssets := initialPoolLiquidity.Len()
-	if numAssets < swaproutertypes.MinPoolAssets {
-		return types.ErrTooFewPoolAssets
-	}
-	if numAssets > swaproutertypes.MaxPoolAssets {
-		return sdkerrors.Wrapf(
-			types.ErrTooManyPoolAssets,
-			"pool has too many PoolAssets (%d)", numAssets,
-		)
-	}
-	return nil
-}
-
-func (k Keeper) validateCreatedPool(
-	ctx sdk.Context,
-	initialPoolLiquidity sdk.Coins,
-	poolId uint64,
-	pool swaproutertypes.PoolI,
-) error {
-	if pool.GetId() != poolId {
-		return sdkerrors.Wrapf(types.ErrInvalidPool,
-			"Pool was attempted to be created with incorrect pool ID.")
-	}
-	if !pool.GetAddress().Equals(types.NewPoolAddress(poolId)) {
-		return sdkerrors.Wrapf(types.ErrInvalidPool,
-			"Pool was attempted to be created with incorrect pool address.")
-	}
-	// Notably we use the initial pool liquidity at the start of the messages definition
-	// just in case CreatePool was mutative.
-	if !pool.GetTotalPoolLiquidity(ctx).IsEqual(initialPoolLiquidity) {
-		return sdkerrors.Wrapf(types.ErrInvalidPool,
-			"Pool was attempted to be created, with initial liquidity not equal to what was specified.")
-	}
-	// This check can be removed later, and replaced with a minimum.
-	if !pool.GetTotalShares().Equal(types.InitPoolSharesSupply) {
-		return sdkerrors.Wrapf(types.ErrInvalidPool,
-			"Pool was attempted to be created with incorrect number of initial shares.")
-	}
-	return nil
-}
-
-// CreatePool attempts to create a pool returning the newly created pool ID or
-// an error upon failure. The pool creation fee is used to fund the community
-// pool. It will create a dedicated module account for the pool and sends the
-// initial liquidity to the created module account.
-//
-// After the initial liquidity is sent to the pool's account, shares are minted
-// and sent to the pool creator. The shares are created using a denomination in
-// the form of gamm/pool/{poolID}. In addition, the x/bank metadata is updated
-// to reflect the newly created GAMM share denomination.
-func (k Keeper) CreatePool(ctx sdk.Context, msg swaproutertypes.CreatePoolMsg) (uint64, error) {
```

Comment:
Note: moved to swaprouter. `InitializePool` now performs gamm-specific pool initializaiton